libff/algebra/curves/bls12_377/bls12_377_g2.cpp

```diff
@@ -388,9 +388,52 @@ bls12_377_G2 bls12_377_G2::mul_by_q() const
                       (this->Z).Frobenius_map(1));
 }
 
+bls12_377_G2 bls12_377_G2::untwist_frobenius_twist() const
```

Comment:
That would be ideal to define this function as the composition of the 3 maps, i.e. have a `twist()` function that "pulls" twisted points back to the curve, `frobenius_endomorphism` (taking a point P(x,y) and returning (x^q, y^q)), and `untwist()` to "push" back to the compact representation group.

`bls12_377_G2::untwist_frobenius_twist()` would be implemented as `bls12_377_G2::untwist_frobenius_twist() { return untwist(frobenius_endo(twist(g)));}`

The overhead related to the various function calls and context switching should be negligible.

(Also, I think having the twisting coeffs (and the untwisting coeffs defined as inverse of the twisting coeffs) would be clearer than using variables named `bls12_377_g2_untwist_frobenius_twist_v` etc.)

Reply:
> (Also, I think having the twisting coeffs (and the untwisting coeffs defined as inverse of the twisting coeffs) would be clearer than using variables named bls12_377_g2_untwist_frobenius_twist_v etc.)

You mean havng them as members of G2?  I was hesitant to do that since it makes it clearer that this is not part of the generic G2 interface, but specific to this curve.  Maybe it's worth considering what the interface should look like once this change is made for bw6.

Reply:
As discussed offline, we will address this over all curves, as part of #35.